components/user-service/src/main/java/edu/jhuapl/dorset/ldapuserservice/LdapUserService.java

```
@@ -0,0 +1,198 @@
+
+public class LdapUserService implements UserService {
+    private static final Logger logger = LoggerFactory.getLogger(LdapUserService.class);
+
+    private Map<String, User> users;
+
+    private String ldapServer;
+    private String ldapSearchBase;
+    private String ldapPassword;
+    private String ldapBindDn;
+    private String ldapFilterAttribute;
+
+    private String[] userAttributes;
+    private DirContext ctx;
+    
+    /**
+     * 
+     * LdapUserService
+     * 
+     */
+    public LdapUserService(Config config) {
+        this.users = new HashMap<String, User>();
+        
+        Config conf = config;
+
+        this.ldapServer = conf.getString("ldap-context.ldapServer");
+        this.ldapSearchBase = conf.getString("ldap-context.ldapSearchBase");
+        this.ldapPassword = conf.getString("ldap-context.ldapPassword");
+        this.ldapBindDn = conf.getString("ldap-context.ldapBindDn");
+        this.ldapFilterAttribute = conf.getString("ldap-context.ldapFilterAttribute");
+
+        String userAttributesStr = conf.getString("ldap-search-controls.userAttributes");
+        userAttributes = userAttributesStr.replaceAll("\"", "").split(",");
+        for (int i = 0; i < userAttributes.length; i++) {
+            userAttributes[i] = userAttributes[i].trim();
+        }
+        this.setUserAttributes(userAttributes);
+  
+        Hashtable<String, String> env = new Hashtable<String, String>();
+        env.put(Context.INITIAL_CONTEXT_FACTORY, "com.sun.jndi.ldap.LdapCtxFactory");
+        env.put(Context.SECURITY_AUTHENTICATION, "simple");
+
+        env.put(Context.PROVIDER_URL, this.ldapServer);
+        env.put(Context.SECURITY_PRINCIPAL, this.ldapBindDn);
+        env.put(Context.SECURITY_CREDENTIALS, this.ldapPassword);
+
+        try {
+            this.ctx = new InitialDirContext(env);
+        } catch (NamingException ex) {
+            logger.info("Error when trying to establish the context.");
```

If this fails, can anything succeed later on

---

Everything later on will fail unless a new context is set (setContext())

---

This should really throw an exception then